client/components/domains/register-domain-step/index.jsx

```diff
@@ -753,6 +753,9 @@ class RegisterDomainStep extends React.Component {
 	};
 
 	showValidationErrorMessage( domain, error ) {
+		if ( this.props.transferInAllowed && includes( [ domainAvailability.MAPPED ], error ) ) {
```

Comment:
Maybe wait for the `transferrable` flag we're going to introduce soon, and add the check here.